vindi-woocommerce-subscriptions.php

```diff
@@ -395,24 +395,35 @@ public function action_links($links)
 		 *
 		 * @return bool
 		 */
-		public function validate_add_to_cart($valid, $product_id, $quantity)
+        public function is_new_product_subscription($product)
         {
-            $cart       = $this->settings->woocommerce->cart;
-			$cart_items = $cart->get_cart();
+            return $product->is_type('subscription') || $product->is_type('variable-subscription');
+        }
 
-			$product = wc_get_product($product_id);
+        function is_cart_item_subscription($cart_item)
+        {
+            $subscriptions_types = array('subscription' , 'variable-subscription');
+            return in_array($cart_item['data']->product_type, $subscriptions_types);
+        }
 
+		public function validate_add_to_cart($valid, $product_id, $quantity)
+        {    
+            $cart       = $this->settings->woocommerce->cart;
+            $cart_items = $cart->get_cart();
+            
+            $product = wc_get_product($product_id);
+            
 			if (empty($cart_items))
 				return $valid;
 
-            if ($product->is_type('subscription')) {
-
+            if ($this->is_new_product_subscription($product)) {
+                
```

Comment:
Esta linha em branco não é necessária.

```suggestion
            if ($this->is_new_product_subscription($product)) {
```

Reply:
Alterações realizadas no commit https://github.com/vindi/vindi-woocommerce-subscriptions/pull/146/commits/5fb9a15365658d663be42116031e61e8c7d038f3